core/src/main/scala/kafka/server/FetchSession.scala

```diff
@@ -353,50 +508,68 @@ class SessionlessFetchContext(val fetchData: util.Map[TopicPartition, FetchReque
   * @param time               The clock to use.
   * @param cache              The fetch session cache.
   * @param reqMetadata        The request metadata.
-  * @param fetchData          The partition data from the fetch request.
+  * @param fetchDataAndError  The partition data and topic ID errors from the fetch request.
+  * @param topicIds           The map from topic name to topic IDs
   * @param isFromFollower     True if this fetch request came from a follower.
   */
 class FullFetchContext(private val time: Time,
                        private val cache: FetchSessionCache,
                        private val reqMetadata: JFetchMetadata,
-                       private val fetchData: util.Map[TopicPartition, FetchRequest.PartitionData],
+                       private val fetchDataAndError: FetchDataAndError,
+                       private val topicIds: util.Map[String, Uuid],
                        private val isFromFollower: Boolean) extends FetchContext {
+  val idErrors = new util.LinkedList(fetchDataAndError.idErrors.values())
   override def getFetchOffset(part: TopicPartition): Option[Long] =
-    Option(fetchData.get(part)).map(_.fetchOffset)
+    Option(fetchDataAndError.fetchData.get(part)).map(_.fetchOffset)
 
   override def foreachPartition(fun: (TopicPartition, FetchRequest.PartitionData) => Unit): Unit = {
-    fetchData.forEach(fun(_, _))
+    fetchDataAndError.fetchData.forEach(fun(_, _))
```

Comment:
When you mention that some callers need to iterate over all partitions like CLUSTER ACTION -- I'm a little confused. I thought the request context was passed into AuthHelper for that.

Reply:
I was referring to the following code. It seems to need to iterate every partition through fetchContext so that the UNKNOWN_TOPIC_OR_PARTITION error code can be added for each partition.

```
      if (authHelper.authorize(request.context, CLUSTER_ACTION, CLUSTER, CLUSTER_NAME)) {
        fetchContext.foreachPartition { (topicPartition, data) =>
          if (!metadataCache.contains(topicPartition))
            erroneous += topicPartition -> errorResponse(Errors.UNKNOWN_TOPIC_OR_PARTITION)
```

Reply:
Ok. I understand now. I think in this case though the expected behavior is to return UNKNOWN_TOPIC_ID if the topic ID is unresolved.

Reply:
Yes, we want to return UNKNOWN_TOPIC_ID for all partitions in this case. The unresolved partitions in this PR could also return the unsupported error code.